packages/superset-ui-legacy-plugin-chart-sankey-loop/src/SankeyLoop.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/* eslint-disable no-param-reassign, no-magic-numbers, sort-keys, babel/no-invalid-this */
+import d3 from 'd3';
```

Comment:
You may only need `d3-selection` here instead of the entire `d3`.
Also `d3-sankey-diagram` 's dependencies are the `d3` micro modules (`v >= 4`). 
Now with `3.5.17`, this plugin will require both `v3` and `v4` which is not great.

Reply:
ya, not sure how to quickly fix the v3/v4 issue. :(

rest should be fixed.